crates/ruma-client-api/src/error.rs

```diff
@@ -27,7 +29,11 @@ mod kind_serde;
 #[non_exhaustive]
 pub enum ErrorKind {
     /// M_FORBIDDEN
-    Forbidden,
+    Forbidden {
+        /// The `WWW-Authenticate` header error message.
+        #[cfg(feature = "unstable-msc2967")]
+        authenticate: Option<AuthenticateError>,
```

Comment:
Now we have the exhaustiveness problem again. The attribute on the enum does not auto-apply to its variants. I guess we have to make the variant non-exhaustive and add a c'tor for it.

Reply:
Oh, this should have been on the other PR.

Reply:
Ah right. I have never see non-exhaustive enum variants :slightly_smiling_face: